disco_aws_automation/disco_vpc.py

```diff
@@ -601,123 +395,69 @@ def destroy(self):
-        """ Delete all security group rules."""
-        security_groups = self.get_all_security_groups_for_vpc()
-        for security_group in security_groups:
-            for rule in security_group.rules:
-                for grant in rule.grants:
-                    if grant.group_id:
-                        src_group = DiscoVPC._find_sg_by_id(security_groups, grant.group_id)
-                        try:
-                            logging.debug(
-                                "revoking %s %s %s %s %s", security_group, rule.ip_protocol, rule.from_port,
-                                rule.to_port, src_group)
-                            self.vpc.connection.revoke_security_group(
-                                group_id=security_group.id,
-                                src_security_group_group_id=src_group.id,
-                                src_security_group_owner_id=src_group.owner_id,
-                                ip_protocol=rule.ip_protocol,
-                                from_port=rule.from_port,
-                                to_port=rule.to_port,
-                            )
-                        except EC2ResponseError:
-                            logging.exception("Skipping error deleting sg rule.")
-
-    def _destroy_security_groups(self):
-        """ Find all security groups belonging to vpc and destroy them."""
-        for security_group in self.get_all_security_groups_for_vpc():
-            if security_group.name != u'default':
-                logging.debug("deleting sg: %s", security_group)
-                self.vpc.connection.delete_security_group(group_id=security_group.id)
-
-    def get_all_security_groups_for_vpc(self):
-        """ Find all security groups belonging to vpc and return them """
-        return self.vpc.connection.get_all_security_groups(filters=self.vpc_filter())
-
-    def _destroy_igws(self):
-        """ Find all gateways belonging to vpc and destroy them"""
-        vpc_attachment_filter = {"attachment.vpc-id": self.vpc.id}
-        # delete gateways
-        for igw in self.vpc.connection.get_all_internet_gateways(filters=vpc_attachment_filter):
-            self.vpc.connection.detach_internet_gateway(igw.id, self.vpc.id)
-            self.vpc.connection.delete_internet_gateway(igw.id)
+        for subnet in self.boto3_ec2.describe_subnets(Filters=[self.vpc_filter()])['Subnets']:
+            self.boto3_ec2.delete_subnet(SubnetId=subnet['SubnetId'])
 
     def _destroy_routes(self):
         """ Find all route_tables belonging to vpc and destroy them"""
-        for route_table in self.vpc.connection.get_all_route_tables(filters=self.vpc_filter()):
-            if not route_table.tags:
-                logging.info("Skipping untagged (default) route table %s", route_table.id)
+        for route_table in self.boto3_ec2.describe_route_tables(Filters=[self.vpc_filter()])['RouteTables']:
+            if len(route_table['Tags']) < 1:
```

Comment:
boto3 exposes ["RouteTables"]["Associations"][0]["Main"] which would be even better check for default route tables.

```
In [103]: ["{} {}".format(rt["RouteTableId"], rt["Associations"][0]["Main"]) for rt in client.describe_route_tables(Filters=[{"Name": "vpc-id", "Values": ['vpc-149f2770']}])["RouteTables"]]
Out[103]: 
['rtb-f5325f91 False',
 'rtb-c4325fa0 False',
 'rtb-c6325fa2 False',
 'rtb-ee325f8a False',
 'rtb-f1325f95 False',
 'rtb-e8325f8c False',
 'rtb-f8325f9c False',
 'rtb-f3325f97 False',
 'rtb-fd325f99 False',
 'rtb-ea325f8e False',
 'rtb-f7325f93 False',
 'rtb-e3325f87 True',
 'rtb-c0325fa4 False']
```